frontend/src/lib/lemon-ui/LemonSelectMultiple/LemonSelectMultiple.tsx

```diff
@@ -1,121 +1,264 @@
+
+    // Reset the selected index when the visible options change
+    useEffect(() => {
+        setSelectedIndex(0)
+    }, [visibleOptions.length])
+
+    useEffect(() => {
+        onInputChange?.(inputValue)
+    }, [inputValue])
+
+    const _onActionItem = (item: string): void => {
+        let newValues = [...values]
+        if (values.includes(item)) {
+            // Remove the item
+            if (mode === 'single') {
+                newValues = []
+            } else {
+                newValues.splice(values.indexOf(item), 1)
+            }
+        } else {
+            // Add the item
+            if (mode === 'single') {
+                newValues = [item]
+            } else {
+                newValues.push(item)
+            }
+
+            setInputValue('')
+        }
+
+        onChange?.(newValues)
+    }
+
+    const _onBlur = (): void => {
+        // We ned to add a delay as a click could be in the popover or the input wrapper which refocuses
+        setTimeout(() => {
+            if (popoverFocusRef.current) {
+                popoverFocusRef.current = false
+                inputRef.current?.focus()
+                _onFocus()
+                return
+            }
+            if (allowCustomValues && inputValue.trim() && !values.includes(inputValue)) {
+                _onActionItem(inputValue.trim())
+            } else {
+                setInputValue('')
+            }
+            setShowPopover(false)
+        }, 100)
+    }
+
+    const _onFocus = (): void => {
+        setShowPopover(true)
+        popoverFocusRef.current = true
+    }
+
+    const _onKeyDown = (e: React.KeyboardEvent<HTMLInputElement>): void => {
+        if (e.key === 'Enter') {
+            e.preventDefault()
+
+            const itemToAdd = visibleOptions[selectedIndex]?.key
+            if (itemToAdd) {
+                _onActionItem(visibleOptions[selectedIndex]?.key)
+            }
+        } else if (e.key === 'Backspace') {
+            if (!inputValue) {
+                e.preventDefault()
+                const newValues = [...values]
+                newValues.pop()
+                onChange?.(newValues)
+            }
+        } else if (e.key === 'ArrowDown') {
+            e.preventDefault()
+            setSelectedIndex(Math.min(selectedIndex + 1, options.length - 1))
+        } else if (e.key === 'ArrowUp') {
+            e.preventDefault()
+            setSelectedIndex(Math.max(selectedIndex - 1, 0))
+        }
+    }
+
+    const prefix = (
+        <>
+            {values.map((value) => {
+                const option = options.find((option) => option.key === value) ?? {
+                    label: value,
+                    labelComponent: null,
+                }
+                return (
+                    <>
+                        <LemonSnack title={option?.label} onClose={() => _onActionItem(value)}>
+                            {option?.labelComponent ?? option?.label}
+                        </LemonSnack>
```

Comment:
Possibly beyond the scope of this PR and a more general issue with Snacks inside Inputs but the border around X changing colour seems off
https://github.com/PostHog/posthog/assets/6685876/26ab2563-3fb5-4350-ada4-6c0e4107d9dd

Reply:
Thats in the TODO list already. But yeah the color is definitely off